project-generator/src/main/java/com/redhat/ipaas/project/converter/DefaultProjectGenerator.java

```diff
@@ -137,9 +136,11 @@ public DefaultProjectGenerator(ConnectorCatalog connectorCatalog, ProjectGenerat
     }
     */
 
-    private byte[] generateFlowYaml(GenerateProjectRequest request) throws JsonProcessingException {
+    private byte[] generateFlowYaml(Map<String, byte[]> contents, GenerateProjectRequest request) throws JsonProcessingException {
         Flow flow = new Flow();
+        int stepCounter[] = new int[]{0};
         request.getIntegration().getSteps().ifPresent(steps -> {
+            stepCounter[0]++;
```

Comment:
Would use a plain int here and declare it within the closure since it is used nowhere outside of the closure.